integration/airflow/openlineage/airflow/listener.py

```diff
@@ -0,0 +1,141 @@
+import logging
+import threading
+import uuid
+import attr
+
+from typing import TYPE_CHECKING, Optional, Callable
+
+from airflow.listeners import hookimpl
+
+from openlineage.airflow.adapter import OpenLineageAdapter
+from openlineage.airflow.extractors import ExtractorManager
+from openlineage.airflow.utils import DagUtils, get_task_location, get_job_name, get_custom_facets
+
+if TYPE_CHECKING:
+    from airflow.models import TaskInstance, BaseOperator
+    from sqlalchemy.orm import Session
+
+
+@attr.s(frozen=True)
+class RunData:
+    run_id: str = attr.ib()
+    task: "BaseOperator" = attr.ib()
+
+
+class RunDataHolder:
+    """Class that stores run data - run_id and task in-memory. This is needed because Airflow
+    does not always pass all runtime info to on_task_instance_success and
+    on_task_instance_failed that is needed to emit events. This is not a big problem since
+    we're only running on worker - in separate process that is always spawned (or forked) on
+    execution, just like old PHP runtime model.
+    """
+    def __init__(self):
+        self.run_data = {}
+
+    def set_run_data(self, task_instance: "TaskInstance", run_id: str):
+        self.run_data[self._pk(task_instance)] = RunData(run_id, task_instance.task)
+
+    def get_run_data(self, task_instance: "TaskInstance") -> Optional[RunData]:
+        pk = self._pk(task_instance)
+        if pk not in self.run_data:
+            return None
+        return self.run_data[pk]
+
+    def _pk(self, ti: "TaskInstance"):
+        return ti.dag_id + ti.task_id + ti.run_id
+
+
+run_data_holder = RunDataHolder()
+extractor_manager = ExtractorManager()
+adapter = OpenLineageAdapter()
+log = logging.getLogger('airflow')
+
+
+def run_thread(target: Callable, kwargs=None):
```

Comment:
_Minor_: I'd use similar naming as underlying method you're calling `start_thread()` (though semantically the same thing)